arch/arm/src/gd32f4/gd32f4xx_sdio.c

```diff
@@ -0,0 +1,3070 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <nuttx/config.h>
+
+#include <inttypes.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <assert.h>
+#include <debug.h>
+#include <errno.h>
+#include <stdio.h>
+
+#include <nuttx/arch.h>
+#include <nuttx/wdog.h>
+#include <nuttx/clock.h>
+#include <nuttx/sdio.h>
+#include <nuttx/wqueue.h>
+#include <nuttx/semaphore.h>
+#include <nuttx/mmcsd.h>
+
+#include <nuttx/irq.h>
+#include <arch/board/board.h>
+
+#include "chip.h"
+#include "gd32f4xx.h"
+#include "gd32f4xx_sdio.h"
+
+#ifdef CONFIG_GD32F4_SDIO
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+/* Configuration ************************************************************/
+
+/* Required system configuration options:
+ *
+ *   CONFIG_ARCH_DMA - Enable architecture-specific DMA subsystem
+ *     initialization.  Required if CONFIG_GD32F4_SDIO_DMA is enabled.
+ *   CONFIG_GD32F4_DMA1 - Enable GD32 DMA1 support.  Required if
+ *     CONFIG_GD32F4_SDIO_DMA is enabled
+ *   CONFIG_SCHED_WORKQUEUE -- Callback support requires work queue support.
+ *
+ * Driver-specific configuration options:
+ *
+ *   CONFIG_GD32F4_SDIO_DMA - Enable SDIO.  This is a marginally optional.
+ *     For most usages, SDIO will cause data overruns if used without DMA.
+ *     NOTE the above system DMA configuration options.
+ *   CONFIG_GD32F4_SDIO_WIDTH_D1_ONLY - This may be selected to force the
+ *     driver operate with only a single data line (the default is to use
+ *     all 4 SD data lines).
+ *   CONFIG_SDM_DMAPRIO - SDIO DMA priority.  This can be selected if
+ *     CONFIG_GD32F4_SDIO_DMA is enabled.
+ *   CONFIG_SDIO_XFRDEBUG - Enables some very low-level debug output
+ *     This also requires CONFIG_DEBUG_FS and CONFIG_DEBUG_INFO
+ */
+
+#ifndef CONFIG_SCHED_WORKQUEUE
+#  error "Callback support requires CONFIG_SCHED_WORKQUEUE"
+#endif
+
+#ifdef CONFIG_GD32F4_SDIO_DMA
+
+#  error "Now SDIO DMA has not ready"
+
+#  ifndef CONFIG_GD32_SDIO_DMAPRIO
+#    if  defined(CONFIG_GD32F4_GD32F4XX)
+#      define CONFIG_GD32_SDIO_DMAPRIO  DMA_PRIORITY_ULTRA_HIGH
+#    else
+#      error "Unknown GD32 DMA"
+#    endif
+#  endif
+#  if defined(CONFIG_GD32F4_GD32F4XX)
+#    if (CONFIG_GD32_SDIO_DMAPRIO & ~DMA_CHXCTL_PRIO_MASK) != 0
+#      error "Illegal value for CONFIG_GD32_SDIO_DMAPRIO"
+#    endif
+#  else
+#    error "Unknown GD32 DMA"
+#  endif
+#else
+#  undef CONFIG_GD32_SDIO_DMAPRIO
```

Comment:
Please remove the code below "#error ", it will not be executed anyway.

Also, please change the error message to:
```
#  error "SDIO DMA Support not implemented yet!"
```